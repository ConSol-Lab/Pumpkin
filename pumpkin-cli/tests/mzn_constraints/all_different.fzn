var 1..4: x1 :: output_var;
var 1..4: x2 :: output_var;
var 1..4: x3 :: output_var;
var 1..4: x4 :: output_var;

constraint pumpkin_all_different([x1, x2, x3, x4]);

solve satisfy;
