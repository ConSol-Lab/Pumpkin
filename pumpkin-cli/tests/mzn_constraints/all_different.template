predicate fzn_all_different_int(array [int] of var int: x);

var 1..4: x1 :: output_var;
var 1..4: x2 :: output_var;
var 1..4: x3 :: output_var;
var 1..4: x4 :: output_var;

constraint fzn_all_different_int([x1, x2, x3, x4]);

solve satisfy;
