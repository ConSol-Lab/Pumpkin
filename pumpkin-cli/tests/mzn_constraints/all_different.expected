x1 = 4;
x2 = 3;
x3 = 2;
x4 = 1;
----------
x1 = 3;
x2 = 4;
x3 = 2;
x4 = 1;
----------
x1 = 4;
x2 = 2;
x3 = 3;
x4 = 1;
----------
x1 = 2;
x2 = 4;
x3 = 3;
x4 = 1;
----------
x1 = 3;
x2 = 2;
x3 = 4;
x4 = 1;
----------
x1 = 2;
x2 = 3;
x3 = 4;
x4 = 1;
----------
x1 = 4;
x2 = 3;
x3 = 1;
x4 = 2;
----------
x1 = 3;
x2 = 4;
x3 = 1;
x4 = 2;
----------
x1 = 4;
x2 = 1;
x3 = 3;
x4 = 2;
----------
x1 = 1;
x2 = 4;
x3 = 3;
x4 = 2;
----------
x1 = 3;
x2 = 1;
x3 = 4;
x4 = 2;
----------
x1 = 1;
x2 = 3;
x3 = 4;
x4 = 2;
----------
x1 = 4;
x2 = 2;
x3 = 1;
x4 = 3;
----------
x1 = 2;
x2 = 4;
x3 = 1;
x4 = 3;
----------
x1 = 4;
x2 = 1;
x3 = 2;
x4 = 3;
----------
x1 = 1;
x2 = 4;
x3 = 2;
x4 = 3;
----------
x1 = 2;
x2 = 1;
x3 = 4;
x4 = 3;
----------
x1 = 1;
x2 = 2;
x3 = 4;
x4 = 3;
----------
x1 = 3;
x2 = 2;
x3 = 1;
x4 = 4;
----------
x1 = 2;
x2 = 3;
x3 = 1;
x4 = 4;
----------
x1 = 3;
x2 = 1;
x3 = 2;
x4 = 4;
----------
x1 = 1;
x2 = 3;
x3 = 2;
x4 = 4;
----------
x1 = 2;
x2 = 1;
x3 = 3;
x4 = 4;
----------
x1 = 1;
x2 = 2;
x3 = 3;
x4 = 4;
----------
==========
